### DISCUSSION TITLE
Add modes to resources

### DISCUSSION THREAD

**Discussion by builder-247** [HypixelDev/PublicAPI#197] [Ideas]
Due to introduction of the `recentGames` endpoint, a new resources for converting the mode values should be added. This data is already available via the unofficial https://api.connorlinfoot.com/v2/games/hypixel/ endpoint.

**Comment by ConnorLinfoot** [comment]
We have something in the works currently which will be released publicly when ready but we want to include as many if not all the modes we can.

I'll leave this issue open so people can subscribe for when its added.

**Reply by ConnorLinfoot** [reply]
I want to give an update on this. We've added a new "early access" endpoint that contains some games + modes. I'll be going through and adding more games over the next few days/weeks. Hopefully, we'll end up with all of them in the end. As a fallback you can also take the mode key, replace the underscores with a space and capitalize each word, this is what we currently do for Game History and Replay's internally that also use this exact same data.

We've also fixed an issue with Murder Mystery mode being wrong when in recent games compared to status, for example, if there are other games that have similar issues that we haven't noticed please let us know. Some way may be able to fix it as long as it's not too intertwined with other systems, this may also mean that some keys in endpoints such as recentgames or status may change to add their prefix, as we seem to have previously removed these.

We're also open to feedback on adding other data (we will not change the structure as to make sure we don't break compatibility with systems using this), some may be easy and thus will be added if requested but some things could be more work so do understand if we don't add everything requested. But I do want to involve developers where possible in adding information.

The endpoint is a resource at https://api.hypixel.net/resources/games and does not require an API key.

**Reply by ConnorLinfoot** [reply]
@mdashlw @builder-247 (not sure if GitHub notifies you for these, so tagging just-in-case since you we're interested)

**Reply by TheNullicorn** [reply]
When "indexing" modes, can they be considered case-insensitive?

**Reply by ConnorLinfoot** [reply]
I'm not sure what you mean by indexing exactly, the mode keys internally are usually case sensitive but it also depends on what it's being used for. You should be safe to ignore casing however as I doubt we currently or ever will have different keys just based on their casing.

**Reply by mdashlw** [reply]
I'm seeing `cleanName`, but don't GameTypes have a short name and a full name? Like `UHC` and `UHC Champions`

**Reply by ConnorLinfoot** [reply]
@mdashlw Clean name is something I just added from the current documentation regarding GameTypes (as well as the ID and database name). We don't actually have a strong defined "short name" at this time internally. It's likely just used on a case-by-case basis, but we could look at hard defining that.

We could also look at defining our aliases, as internally we do have them defined for some games which could be useful for developers.

**Reply by mdashlw** [reply]
Yeah, it would be very helpful to have more information than the documented GameType enum. I also think `name` would be better than `cleanName`, because a) if `shortName` is added, it would be consistent (it won't be `cleanShortName`); b) consistency with `modeNames` (it's not `modeCleanNames`)

**Reply by ConnorLinfoot** [reply]
That's true, I'll quickly change that now since it's only been 30 minutes so i hope nobody (or few) are using those fields already.

**Reply by mdashlw** [reply]
Some GameType names are different from what I collected from the server (https://gist.github.com/mdashlw/fa587a74e8bd93013656c6a146fb476f), e.g. `ARCADE`, `TNTGAMES`, `PIT`. Can you double-check the resource?

**Reply by ConnorLinfoot** [reply]
Yeah, for some reason we seem to have different display names in different places. I took the names from the current OpenAPI documentation but that conflicts with internal names and the GameType enum in the Java implementation. I will look at updating this tomorrow.

**Reply by tfecloud** [reply]
What about fishing does that count to the main lobby, if yes which id does the main lobby have 1?

**Comment by mdashlw** [comment]
It has been two months, is there any update on this? I also noticed game mode name varies on context, in /status: `MURDER_CLASSIC`, in /recentGames: just `CLASSIC`, making it even more difficult to make the whole mode list by ourselves
